src/LodeRunner.API/src/Data/Cache/LRAPICache.cs

```diff
@@ -0,0 +1,213 @@
+             .RegisterPostEvictionCallback(async (key, value, reason, state) =>
+             {
+                 // log the request
+                 logger.LogInformation(nameof(LRAPICache), "Cache data request.");
+
+                 // NOTE: EvictionReason.Removed or EvictionReason.Replaced
+                 if (reason <= EvictionReason.Replaced)
+                 {
+                     // NOTE: Scenario LRAPI collection has pending feed change that Calls ProcessClientStatusChange(), then it triggers a key replace event.
+                     // Also if we already initiated Cache.Set() to do a replace we do not need to query and replace the key,
+                     // neither if key was removed for a EvictionReason other that  Expired,  TokenExpired  or  Capacity.
+                     // if so  this will cause a indefinitely loop, since we call this.Cache.Set to update the key.
+
+                     return;
+                 }
+
+                 string clientStatusId = key.ToString();
+                 try
+                 {
+                     // Get Client Status from Cosmos
+                     ClientStatus clientStatus = await clientStatusService.Get(clientStatusId);
+
+                     // if still exists, update
+                     this.SetEntry(key, new Client(clientStatus), GetMemoryCacheEntryOptions());
+                 }
+                 catch (CosmosException ce)
+                 {
+                     // log Cosmos status code
+                     if (ce.StatusCode == HttpStatusCode.NotFound)
+                     {
+                         logger.LogInformation("MemoryCacheEntryOptions.RegisterPostEvictionCallback", $"{logger.NotFoundError}: Removing Client {clientStatusId} from Cache");
+                     }
+                     else
+                     {
+                         logger.LogError("MemoryCacheEntryOptions.RegisterPostEvictionCallback", ce.ActivityId, new LogEventId((int)ce.StatusCode, "CosmosException"), ex: ce);
+                     }
+                 }
+                 catch (Exception ex)
+                 {
+                     // log exception
+                     logger.LogError("MemoryCacheEntryOptions.RegisterPostEvictionCallback", "Exception", NgsaLog.LogEvent500, ex: ex);
+                 }
+             });
+        }
+
+        private void SetClientCache()
+        {
+            // log the request
+            logger.LogInformation(nameof(LRAPICache), "Cache data request");
+
+            try
+            {
+                // cache client statuses
+                var results = clientStatusService.GetAll().Result.ToList();
+
+                foreach (var item in results)
+                {
+                    var client = new Client(item);
+
+                    this.SetEntry(client.ClientStatusId, client, GetMemoryCacheEntryOptions());
+                }
+            }
+            catch (CosmosException ce)
```

Comment:
It looks like we swallow the error here, but since this cache is required as the access to ClientStatus for LRAPI we need to bubble the error up, because the app should exit as it won't be able to operate.

Reply:
fixed